### ISSUE TITLE
2019-11-25 tc014 M2SS - OPTIMIZEKEL causes computation error when adding TLASTi to PARAMETERLIST

### ISSUE THREAD

**Issue by jhhughes256** [tensfeldt/openNCA#58]
### Error

When running OPTIMIZEKEL for tc014 (by adding "TLASTi" to the original parameterset) a computation error occurs:

```
Error in `[<-.data.frame`(`*tmp*`, , map_data$FLGEXKEL, value = c(1, 1,  : 
  replacement has 135 rows, data has 161 
4. stop(sprintf(ngettext(N, "replacement has %d row, data has %d", 
    "replacement has %d rows, data has %d"), N, nrows), domain = NA) 
3. `[<-.data.frame`(`*tmp*`, , map_data$FLGEXKEL, value = c(1, 1, 
0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 0, 
0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 
1, 1, 1, 0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 0,  ... at run_computation.R#2280
2. `[<-`(`*tmp*`, , map_data$FLGEXKEL, value = c(1, 1, 0, 0, 0, 
0, 1, 1, 1, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 0, 0, 1, 0, 
1, 1, 0, 0, 0, 0, 0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 1, 1, 1, 
0, 0, 0, 1, 1, 0, 1, 0, 0, 1, 1, 1, 0, 0, 0, 1, 1, 0, 0, 1, 1,  ... at run_computation.R#2280
1. run_computation(data = d, map = mct, flag = flags, parameterset = parameterset) 
```

As mentioned in #2 the `parameterset` input for tc011, tc014 and tc071 did not request the required variables to activate OPTIMIZEKEL. This is solved by ensuring that "TMAXi", 'CMAXi", "TLASTi", "CLASTi" and "AUCLASTi" are provided by the `parameterset`. When adding the missing "TLASTi" to tc014 the error above occurs.

### Example

To replicate this error, tc014 was loaded and the following changes were made:

```
mct$OPTIMIZEKEL <- 1
mct$PARAMETERLIST <- paste(mct$PARAMETERLIST, "TLASTi", sep = ";")
```

This error is not impacted by changes to FLGEXKEL in the `flags` input or FLGNOCMAX in the `mct`. The parameterset is displayed below:

```
> stringr::str_split(mct$PARAMETERLIST, ";")[[1]]
 [1] "AUCALL"      "AUCDN"       "AUCINFO"     "AUCINFOC"    "AUCINFODN"   "AUCINFP"     "AUCINFPC"   
 [8] "AUCINFPDN"   "AUCLASTi"    "AUCLASTDNi"  "AUCLASTCi"   "AUCT"        "AUCT1_T2"    "AUCTAUi"    
[15] "AUCTAUDN"    "AUCXBOCTO"   "AUCXBPCTP"   "AUCXPCTO"    "AUCXPCTP"    "AUMCTAUi"    "C0"         
[22] "CAVi"        "CLASTi"      "CLRTAUi"     "CLTAUi"      "CLTAUWi"     "CMAXi"       "CMAXCi"     
[29] "CMAXDNi"     "CMINi"       "DOSEC"       "CTROUGHi"    "CTROUGHENDi" "DIi"         "DOSEi"      
[36] "FREL"        "FRELLAST"    "FTAUi"       "KEL"         "KELNOPT"     "KELRSQ"      "KELRSQA"    
[43] "KELTMHI"     "KELTMLO"     "LASTTIME"    "MRAUCINF"    "MRAUCLAST"   "MRAUCTAUi"   "MRCMAXi"    
[50] "MRTIVIFOi"   "MRTIVIFPi"   "PTFi"        "PTRi"        "PTROUGHRi"   "PTROUGHREND" "TAU1"       
[57] "TAUi"        "THALF"       "THALFF"      "TMAXi"       "TMINi"       "TOLD1"       "TOLD"       
[64] "V0"          "VSSPi"       "VSSO"        "VSSOW"       "VSSPWi"      "VZTAUi"      "TLASTi"  
```

**Comment by tensfeldt** [comment]
This is not a correct behavior. TLASTi must be added to the create_dependency_list.R code to accommodate. We cannot change the input simply because the optimize KEL routine needs it. Please ensure that the optimize kel routine uses the same methods as the other parameter calculations by casting its required dependencies into the dependency list.

**Comment by jhhughes256** [comment]
### Update

This issue still occurs as of 18821dca00. "TLASTi" no longer needs to be added to make OPTIMIZEKEL run as per the fix for #2, but the error still occurs when running tc014 with OPTIMIZEKEL == 1.

**Comment by jhhughes256** [comment]
This issue is likely related to the sparse data from this example. I'm not sure if we can expect OPTIMIZEKEL to work with such a small number of points.

```
   0    1   24  504  672 <NA> 
  32   32   22   14    6   55 
```

**Comment by rudraya** [comment]
This issue was addressed in 6e99fab, please let me know if it still occurs.

**Comment by tensfeldt** [comment]
While the results of the OPTIMIZEKEL routine was not tested thoroughly, when TLASTi is added to PARAMETERLIST for tc014 M2SS, the OPTIMIZEKEL routine completes successfully and chooses different FLGEXKEL values than were available in the input FLAGS dataset. This issue can be closed for this reason as of testing with commit b8deb76
